src/Sql/Insert.php

```diff
@@ -160,7 +160,8 @@ public function getRawState($key = null)
         $rawState = [
             'table' => $this->table,
             'columns' => array_keys($this->columns),
-            'values' => array_values($this->columns)
+            'values' => array_values($this->columns),
```

Comment:
No, it does not prevent it. It actually helps.
This PR will not break any existing code.
However, I can see yet another bug report in sequence feature area in few weeks/month time with someone claiming a bug with a usecase I posted in the issue discussion.

I tend to get too verbose and sometimes my point get lost in the wall of text (to which this sentence ironically contributes) so will try to highlight the issue again.

To re-emphasize ,your code so far does not break anything, but is *incomplete*, if *filling out columns key is left out*. It can technically be accepted, but I personally do not feel it will be appropriate since that sequence thing is another bug report just waiting to happen and it can be prevented if this PR is expanded on to allow for a facility to fully inspect column values with SELECT the same way it is done with manual column specification.